### ISSUE TITLE
Plans for supporting powershell core / 6.1?

### ISSUE THREAD

**Issue by Larswa** [chocolatey/boxstarter#353]
Hi, Thanks for a great tool!  

I was wondering if there was any plan for supporting powershell core 6.1? I have started using it on a daily basis and are tired of having to fall back to 5.1 for using boxstarter. 

Importing the modules in pwsh 6.1 works fine, but some commandlets use the deprecated get-wmiobject.
See about deprecation here: https://docs.microsoft.com/en-us/powershell/scripting/whats-new/breaking-changes-ps6?view=powershell-6

For instance - the Enable-Rdp commandlet from the winconfig module could probably quite easily be rewritten to use the newer CIM commandlets instead. So I was thinking this could be something you would take pullrequests for implemeting? If it is on the roadmap?  Or maybe just the commands that I need to use in PS6?

best regards
Lars

**Comment by pauby** [comment]
@Larswa What is your use case that you are using this on a daily basis? As Boxstarter is aimed at building Windows machines from scratch and Windows ships with Windows PowerShell I'm trying to understand the benefit of working with PowerShell Core _at the moment_.

**Comment by pauby** [comment]
@Larswa I get where you are coming from but I really don't see it as a priority at the moment given the prevalence of Windows PowerShell on the machines Boxstarter is aimed at.

> but PS 6.x is what we are all going to use going forward

That's a huge generalisation I disagree with entirely. I use PowerShell Core as my daily driver but I recognise that I am in the minority and that I will have issues with it for the time being (given it only supports a subset of Windows PowerShell compatibility).

> Looking at the Boxstarter modules, I don't think it would be too hard to make them both 6.1 and 5.1 compatible? Although I havent been into all corners :-)

I think you're right in that we could support the majority of it through PS Core. It's something that we'd need to take time to do and as I said not a priority at the moment.

> But if 6.x is not on the radar, yet, I'll just bootstrap everything with 5.1 untill now, and keep switching to 6.1 once installation is done.

I think that's the best course. As Windows PowerShell comes out of the box you'd need to install either Chocolatey to install PS Core (and therefore use Boxstarter anyway) or do it manually. At the end of the build though you're going to have PS Core which is what is still the same outcome.

**Comment by Larswa** [comment]
@pauby Thanks for your time, and I won't Bug you any futher, the issue but I would speculate on this; I would suspect that with the higher release cadence of Windows client and server, and the fact that no new features are coming to WMF 5.1, that we will probably see some some new Windows features that are only out of the box supported by PS 6x?

**Comment by pauby** [comment]
@Larswa that we will probably see some some new Windows features that are only out of the box supported by PS 6x?

Absolutely. I am wondering what they are going to do. If they consider Windows PowerShell 'feature complete' then what are they going to add to PS Core?!?!?!

**Comment by Larswa** [comment]
@pauby  Good question. Not sure how I feel about the cross platform Mac/*nix compatibility thing. I have a hard time seeing that it will catch on when up against a solid native unix bash userbase. 

Where I see PS6 as a good bet going forward  is the unlocking from a specific windows version. Who nows where they will take windows, but I rest well at night knowing that Powershell will go where windows goes. But this is all pretty esoteric and pure speculation right now. Still ... the kids are going to have to learn powershell right after they have learnt the alphabet ...  :-D    Get-Oatmeal | Pour-Milk ...

**Comment by philcarbone** [comment]
Powershell core will be shipped with windows (in v7) - FYI

> We are planning on eventually shipping PowerShell 7 in Windows as a side-by-side feature with Windows PowerShell 5.1, but we still need to work out some of the details on how you will manage this inbox version of PowerShell 7.

https://devblogs.microsoft.com/powershell/the-next-release-of-powershell-powershell-7/

.NET Core is now replacing .NET Framework as .NET 5 (and this affects support for Powershell as it's based on .NET Framework)
https://devblogs.microsoft.com/dotnet/introducing-net-5/
